### PR TITLE
feat(vue 3): add search

### PR DESCRIPTION
**Pull request by juliodialpad**
## Description

We need to add docsearch to VuePress site too, I'm adding it with the default styling for now, we had a input to type in before, but docsearch@3 is using a button

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [ ] Ensure all `gulp` scripts successfully compile.
 - [ ] Update, remove, or extend all affected documentation.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
 - [x] Request a review from Brad Paugh, David Becher, or Drew Chandler.

### CODE REVIEW

File: docs/.vuepress/config.ts [hunk 1]
```diff
@@ -143,6 +143,9 @@ export default defineUserConfig<DefaultThemeOptions>({
       content: "Dialtone is the design system and resources for the Dialpad team."
     }],
 
+    // Docsearch
+    ['meta', {name: "docsearch:version", content: 'vuepress'}],
```
Comment (braddialpad) [inline docs/.vuepress/config.ts]: don't need this anymore now that we added the exclude folders

File: docs/.vuepress/theme/components/Navbar.vue [hunk 2]
```diff
@@ -40,19 +40,13 @@
...
         </a> -->
       </div>
-      <div class="d-ps-relative js-input-search-wrapper md:d-d-none">
-        <span class="d-input-icon d-input-icon--left"></span>
-        <input class="d-input d-pl32 js-dialtone-search" placeholder="Search Dialtone" type="search"/>
-        <div class="d-ps-absolute d-t6 d-l4 d-ml1 d-fc-black-600">
-        </div>
-      </div>
-      <!--      {% include mobile-header.html %}-->
+      <div id="algolia-search-container" />
```
Comment (braddialpad) [inline docs/.vuepress/theme/components/Navbar.vue]: Is this just using default styling?

File: docs/.vuepress/theme/components/Navbar.vue [hunk 3]
```diff
@@ -62,5 +56,26 @@ export default {
...
+        import(/* webpackChunkName: "docsearch" */ '@docsearch/css'),
+      ]).then(([docsearch]) => {
+        docsearch = docsearch.default
+        docsearch({
+          indexName: 'dialpad',
+          apiKey: '6436ebddb959748daeec411eb388a99d',
+          container: '#algolia-search-container',
+          appId: 'Y5HG9UX6KM',
+          placeholder: 'Search Dialtone',
+          facetFilters: ['version:vuepress']
```
Comment (braddialpad) [inline docs/.vuepress/theme/components/Navbar.vue]: can remove facetfilters now that we added exclude folders

File: docs/.vuepress/theme/components/Navbar.vue [hunk 4]
```diff
@@ -62,5 +56,26 @@ export default {
     brandLogo: useThemeLocaleData().value.logo,
     brandLogoAlt: useSiteLocaleData().value.title
   }),
+  mounted() {
+    this.initialize()
```
Comment (braddialpad) [inline docs/.vuepress/theme/components/Navbar.vue:60]: Does this need to reinitialize every time a new page is loaded or does it only happen once?

**Comment by braddialpad** [comment]
> 

lol 🦑 😆

File: docs/.vuepress/theme/components/Navbar.vue (hunk 4 above)
Comment (juliodialpad) [reply docs/.vuepress/theme/components/Navbar.vue:60]: It just reinitialize once you change the main Layout, like from HomePage to getting started, but while you are navigating the site it doesn't reinitialize until you go to HomePage again

File: docs/.vuepress/theme/components/Navbar.vue (hunk 2 above)
Comment (juliodialpad) [reply docs/.vuepress/theme/components/Navbar.vue]: Yeah, should I implement the styling you added to it?

File: docs/.vuepress/theme/components/Navbar.vue (hunk 4 above)
Comment (juliodialpad) [reply docs/.vuepress/theme/components/Navbar.vue:60]: > It just reinitialize once you change the main Layout, like from HomePage to getting started, but while you are navigating the site it doesn't reinitialize until you go to HomePage again

Forget about it, found a fix and now it just initializes once.

File: docs/.vuepress/theme/components/Navbar.vue (hunk 2 above)
Comment (braddialpad) [reply docs/.vuepress/theme/components/Navbar.vue]: Hmm yeah might as well

File: docs/.vuepress/theme/components/Navbar.vue (hunk 4 above)
Comment (braddialpad) [reply docs/.vuepress/theme/components/Navbar.vue:60]: Nice

**Review by braddialpad — APPROVED** [review]
Looks good thanks!

**Pull request merged**